core/src/main/scala/pureconfig/ConfigCursor.scala

```diff
@@ -95,6 +95,18 @@ sealed trait ConfigCursor {
   def asMap: Either[ConfigReaderFailures, Map[String, ConfigCursor]] =
     asObjectCursor.right.map(_.map)
 
+  final def atPath(path: Path): Either[ConfigReaderFailures, ConfigCursor] = {
+    path.segments match {
```

Comment:
In the general case, monads such as `Either` are not tail-recursive, so you can't easily make this stack safe. Can you revert this to the old `foldLeft` formulation?